File: carbonaudit/src/main/java/com/carbonaudit/carbonaudit/service/VendorService.java
package com.carbonaudit.carbonaudit.service;

import com.carbonaudit.carbonaudit.dto.VendorDTO;

public interface VendorService {

    Object addVendor(VendorDTO vendorDTO);

    Object getAllVendors();

    Object getVendorById(Long id);

    Object updateVendor(Long id, VendorDTO vendorDTO);

    void deleteVendor(Long id);

}

File: carbon-audit-frontend/src/services/vendorService.js
import api from "../utils/axiosConfig";

export const getAllVendors = async () => {

  const response = await api.get(
    "/vendors"
  );

  return response.data;
};

export const addVendor = async (
  vendorData
) => {

  const response = await api.post(
    "/vendors",
    vendorData
  );

  return response.data;
};

export const deleteVendor = async (
  id
) => {

  const response = await api.delete(
    `/vendors/${id}`
  );

  return response.data;
};

const vendorService = {

  getAllVendors,

  addVendor,

  deleteVendor

};

export default vendorService;